prov/slurm/execution.go

```diff
@@ -426,25 +428,34 @@ func (e *executionCommon) prepareAndSubmitJob(ctx context.Context) error {
 	var cmd string
 	if e.jobInfo.Command != "" {
 		inner := fmt.Sprintf("%s %s", e.jobInfo.Command, quoteArgs(e.jobInfo.Args))
-		cmd = e.wrapCommand(inner)
+		var err error
+		cmd, err = e.wrapCommand(inner)
+		if err != nil {
+			return err
+		}
 	} else {
 		cmd = fmt.Sprintf("%s%ssbatch -D %s%s %s", e.addWorkingDirCmd(), e.buildEnvVars(), e.jobInfo.WorkingDir, e.buildJobOpts(), path.Join(e.jobInfo.WorkingDir, e.PrimaryFile))
 	}
 	return e.submitJob(ctx, cmd)
 }
 
-func (e *executionCommon) wrapCommand(innerCmd string) string {
-	pathScript := path.Join(e.jobInfo.WorkingDir, batchScript)
+func (e *executionCommon) wrapCommand(innerCmd string) (string, error) {
+	id, err := uuid.NewRandom()
+	if err != nil {
+		return "", errors.Wrap(err, "failed to generate UUID for generated slurm batch script name")
+	}
+	scriptName := fmt.Sprintf(batchScript, id.String())
+	pathScript := path.Join(e.jobInfo.WorkingDir, scriptName)
 	// Add the script to the artifact's list
-	e.jobInfo.Artifacts = append(e.jobInfo.Artifacts, batchScript)
+	e.jobInfo.Artifacts = append(e.jobInfo.Artifacts, scriptName)
 	// Write script
 	cat := fmt.Sprintf(`cat <<'EOF' > %s
 #!/bin/bash
 
 %s
 EOF
 `, pathScript, innerCmd)
-	return fmt.Sprintf("%s%s%ssbatch -D %s%s %s", e.addWorkingDirCmd(), e.buildEnvVars(), cat, e.jobInfo.WorkingDir, e.buildJobOpts(), pathScript)
+	return fmt.Sprintf("%s%s%ssbatch -D %s%s %s; rm %s", e.addWorkingDirCmd(), e.buildEnvVars(), cat, e.jobInfo.WorkingDir, e.buildJobOpts(), pathScript, pathScript), nil
```

Comment:
may use `/bin/rm %s` or `rm -f` just in case `rm` is aliases to `rm -i` on the host

Reply:
Yep good idea